### PR TITLE
[BOLT] CDSplit main logic part 1/2

### PR DESCRIPTION
**Pull request by ShatianWang**
This diff defines and initializes auxiliary variables used by CDSplit and implements two important helper functions. The first helper function approximates the block level size increase if a function is hot-warm split at a given split index (X86 specific). The second helper function finds all calls in the form of X->Y or Y->X for each BF given function order [... X ... BF ... Y ...]. These calls are referred to as "cover calls". Their distance will decrease if BF's hot fragment size is further reduced by hot-warm splitting. NFC.

### CODE REVIEW

File: bolt/lib/Passes/SplitFunctions.cpp [hunk 1]
```diff
@@ -173,6 +184,260 @@ struct SplitCacheDirected final : public SplitStrategy {
...
+  DenseMap<const BinaryBasicBlock *, size_t> BBSizes;
+  DenseMap<const BinaryBasicBlock *, size_t> BBOffsets;
+  DenseMap<const BinaryBasicBlock *, JumpInfo> JumpInfos;
+
+  // Sizes of branch instructions used to approximate block size increase
+  // due to hot-warm splitting. Initialized to be 0. These values are updated
+  // if the architecture is X86.
+  uint8_t BRANCH_SIZE = 0;
+  uint8_t LONG_UNCOND_BRANCH_SIZE_DELTA = 0;
+  uint8_t LONG_COND_BRANCH_SIZE_DELTA = 0;
```
Comment (maksfb) [inline bolt/lib/Passes/SplitFunctions.cpp:211]: Is there a reason for these variables to use a different naming scheme?

Comment (maksfb) [reply bolt/lib/Passes/SplitFunctions.cpp:211]: Let's use camel case to follow LLVM coding standards.

Comment (ShatianWang) [reply bolt/lib/Passes/SplitFunctions.cpp]: Just made the changes in [fixup! [BOLT] CDSplit main logic part 1/2](https://github.com/llvm/llvm-project/pull/73895/commits/e07254ef57e3a1db7c404527f32c500b097629b8)

**Review by maksfb — APPROVED** [review]
LGTM

**Pull request merged**